buildscripts/buildRelease.sh

```diff
@@ -359,92 +328,150 @@ cp -p -r $WORKSPACE/KeppelThemeFiles/* $WORKSPACE/src
+    git add dist/ bower.json
+    git commit -m "Committing changes for package $NextVersion-test. Latest commit ID is $latestCommitID."
+    git push --set-upstream origin $NextVersion-package-test
+    popd; popd
 fi
 
 # Return to the develop branch and discard changes to a couple of files
 echo
 echo Returning to the develop branch
-cd $WORKSPACE
 git checkout docs/app/data/footer-navigation.json
 git checkout docs/app/data/landing-page.json
 
+if [ "$BuildPackages" == "true" ]; then
+    # Loop while waiting for the $NextVersion-package-test branch to be merged into the Bower
+    # branch or deleted. Merging signals that the release is verified and so the NPM package may
+    # be created and published.
+    branchCheckDelay=30    
+    while :
+    do
+        branchExists=`git ls-remote --heads https://github.com/UXAspects/UXAspects.git $NextVersion-package-test | wc -l`
+        if [ $branchExists == 0 ] ; then
+            latestBowerCommitID=`git rev-parse origin/bower`
+            echo latestBowerCommitID is $latestBowerCommitID
+            latestBowerCommitMessage=`git log --format=%s%b -n 1 $latestBowerCommitID`
+            echo latestBowerCommitMessage is $latestBowerCommitMessage
+            branchMerged=`echo "$latestBowerCommitMessage" | grep "$NextVersion-package-test" | wc -l`
+            # If this contains $NextVersion-package-test, the branch was merged
+            if [ $branchMerged == 1 ] ; then
+                # Create the NPM package, including a valid authentication token
+                echo Branch $NextVersion-package-test has been merged into the Bower branch. Create the NPM package.
+                echo
+                echo Creating the NPM package
+                if [ -d "$WORKSPACE/npm" ]; then
+                    echo "Folder $WORKSPACE/npm exists... deleting it!"
+                    rm -rf $WORKSPACE/npm
+                fi
+                mkdir -p $WORKSPACE/npm
+                pushd $WORKSPACE/npm
+                cp -p -r $WORKSPACE/src/package.json .
+                cp -p -r $WORKSPACE/dist .
+                # Write the token to the .npmrc file
+                echo "//registry.npmjs.org/:_authToken="$NPMUserPassword > .npmrc
+                # Create a .npmignore file to prevent the .npmrc file ending up in the package
+                echo -e ".npmrc\n.npmignore" > .npmignore
+                
+                # Publish the package
+                echo Publishing the NPM package
+                docker run --rm \
+                    --volume "$PWD":/workspace:rw --workdir /workspace \
+                    --user $UID:$GROUPS \
+                    -e "http_proxy=$HttpProxy" \
+                    -e "https_proxy=$HttpsProxy" \
+                    -e "no_proxy=localhost, 127.0.0.1" \
+                    $UX_ASPECTS_BUILD_IMAGE_NAME:$UX_ASPECTS_BUILD_IMAGE_TAG_LATEST \
+                    npm publish --access public
+                echo NPM package published
```

Comment:
Looking through the code I can see some lines which probably wouldn't work within a container e.g.
SSHing to the Selenium Grid Hub machine. This required the installation of the public key of the build machine user on the Hub machine, That user does not exist within the container so I don't think the remote connection would work.

Also, since the build machine user does not exist within the container, 'git commit' commands could fail as the container user would not have write access rights to the repository.